src/components/Main/Search.js

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useEffect } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import useHttpRequest from '../../hooks/useHttpRequest';
+
+const Search = () => {
+  const navigate = useNavigate();
+
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const {
+    fetchRequest,
+    state: { error, data, loading }
+  } = useHttpRequest({
+    method: 'GET',
+    url: `products/?name=${searchTerm}`,
+    preventAutoFetch: true
+  });
+
+  const search = () => {
+    if (searchTerm.length > 0) {
+      fetchRequest();
+    }
+  };
+
+  useEffect(() => {
+    search();
```

Comment:
Can you add here some timeout to not call fetch request immediately after letter is printed? You can use lodash/debounse function for it

Reply:
Fixed. Thanks!